### PR TITLE
[Bug] Version bump with previous

### LINKED ISSUE #1868: [Bug] Version locking does not bump when forked
https://github.com/elastic/detection-rules/blob/46c2383e5b1247ac6c7e41aea4f130beaf7ec57c/etc/version.lock.json#L1385-L1396

When a rule is forked (existing rule had `min_stack_version` increased), the `previous` entry gets a lock on the bump, but the present version does not get incremented before saving.

The 1.0.2 tagged branch needs to be regenerated after this fix, since it include erroneous versions

## Details

* [PR](https://github.com/elastic/detection-rules/pull/1768/files) where the error occurred (possibly not the first, could be earilier).
* corresponding action [run](https://github.com/elastic/detection-rules/runs/5147618953?check_suite_focus=true)

## 🐛 found

https://github.com/elastic/detection-rules/blob/46c2383e5b1247ac6c7e41aea4f130beaf7ec57c/detection_rules/version_lock.py#L140

Need to update top level with

```python
existing_rule_lock.update(current_rule_lock)
```

Also going to add more verbose logging

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
resolves #1868 

## Summary
When a forked rule has changes, also save version changes to the top level

This also reconciles the errors in the `version.lock.json`

<details>

```
[+] Building package 8.2
 - 13 rules excluded from package
Rule changes detected!
 - 54 changed rules
 - 18 new rules
 - 2 newly deprecated rules
Detailed changes: 
  A: 3115bd2c-0baa-4df0-80ea-45e474b5ef93, new version: 3
    - min_stack_version added: 7.15.0
  A: 5beaebc1-cc13-4bfc-9949-776f9e0dc318, new version: 8
  A: 536997f7-ae73-447d-a12d-bff1e8f5f0a0, new version: 3
  A: 78d3d8d9-b476-451d-a9e0-7a5addd70670, new version: 9
  A: dca28dee-c999-400f-b640-50a081cc0fd1, new version: 9
  A: 8b64d36a-1307-4b2e-a77b-a0027e4d27c8, new version: 4
  A: f766ffaf-9568-4909-b734-75d19b35cbf4, new version: 2
  A: bc48bba7-4a23-4232-b551-eca3ca1e3f20, new version: 6
  A: 04c5a96f-19c5-44fd-9571-a0b033f9086f, new version: 2
  A: 5663b693-0dea-4f2e-8275-f1ae5ff2de8e, new version: 7
  A: ec8efb0c-604d-42fa-ac46-ed1cfbc38f78, new version: 3
  A: ff4dd44a-0ac6-44c4-8609-3f81bc820f02, new version: 6
  A: 0e52157a-8e96-4a95-a6e3-5faae5081a74, new version: 2
  A: 514121ce-c7b6-474a-8237-68ff71672379, new version: 6
  A: 88671231-6626-4e1b-abb7-6e361a171fbb, new version: 2
  A: 97a8e584-fd3b-421f-9b9d-9c9d9e57e9d7, new version: 2
  A: cdbebdc1-dc97-43c6-a538-f26a20c0a911, new version: 1
  A: 8fed8450-847e-43bd-874c-3bbf0cd425f3, new version: 1
  A: 10754992-28c7-4472-be5b-f3770fd04f2d, new version: 1
  A: e9b4a3c7-24fc-49fd-a00f-9c938031eef1, new version: 1
  A: 1859ce38-6a50-422b-a5e8-636e231ea0cd, new version: 1
  A: 0968cfbd-40f0-4b1c-b7b1-a60736c7b241, new version: 1
  A: ee619805-54d7-4c56-ba6f-7717282ddd73, new version: 1
  A: 72d33577-f155-457d-aad3-379f9b750c97, new version: 1
  A: 6f683345-bb10-47a7-86a7-71e9c24fb358, new version: 1
  A: da986d2c-ffbf-4fd6-af96-a88dbf68f386, new version: 1
  A: 83b2c6e5-e0b2-42d7-8542-8f3af86a1acb, new version: 1
  A: 97da359b-2b61-4a40-b2e4-8fc48cf7a294, new version: 1
  A: 89583d1b-3c2e-4606-8b74-0a9fd2248e88, new version: 1
  A: 8da41fc9-7735-4b24-9cc6-c78dfc9fc9c9, new version: 2
  A: 9d19ece6-c20e-481a-90c5-ccca596537de, new version: 4
  A: c292fa52-4115-408a-b897-e14f684b3cb7, new version: 5
  A: 48d7f54d-c29e-4430-93a9-9db6b5892270, new version: 3
  A: 6d448b96-c922-4adb-b51c-b767f1ea5b76, new version: 10
  A: 2f2f4939-0b34-40c2-a0a3-844eb7889f43, new version: 5
  A: bd2c86a0-8b61-4457-ab38-96943984e889, new version: 4
  A: 959a7353-1129-4aa7-9084-30746b256a70, new version: 3
  A: 9f962927-1a4f-45f3-a57b-287f2c7029c1, new version: 2
  A: 208dbe77-01ed-4954-8d44-1e5751cb20de, new version: 1
  A: a4c7473a-5cb4-4bc1-9d06-e4a75adbc494, new version: 1
    - min_stack_version added: 7.15.0
  A: 54c3d186-0461-4dc3-9b33-2dc5c7473936, new version: 3
  A: 577ec21e-56fe-4065-91d8-45eb8224fe77, new version: 6
  A: eb610e70-f9e6-4949-82b9-f1c5bcd37c39, new version: 2
  A: 850d901a-2a3c-46c6-8b22-55398a01aad8, new version: 1
    - min_stack_version added: 7.15.0
  A: 0f93cb9a-1931-48c2-8cd0-f173fd3e5283, new version: 3
    - min_stack_version added: 7.14.0
  A: d117cbb4-7d56-41b4-b999-bdf8c25648a0, new version: 3
  A: f874315d-5188-4b4a-8521-d1c73093a7e4, new version: 4
  A: 2c17e5d7-08b9-43b2-b58a-0270d65ac85b, new version: 7
  A: fe794edd-487f-4a90-b285-3ee54f2af2d3, new version: 3
  A: feeed87c-5e95-4339-aef1-47fd79bcfbe3, new version: 2
  A: 81fe9dc6-a2d7-4192-a2d8-eed98afc766a, new version: 3
  B: 2e29e96a-b67c-455a-afe4-de6183431d0d, new version: 4
    - previous 7.13.0 saved as version: 3
    - current min_stack updated to 7.15
  A: f63c8e3c-d396-404f-b2ea-0379d3942d73, new version: 4
  A: ad0d2742-9a49-11ec-8d6b-acde48001122, new version: 1
  A: eda499b8-a073-4e35-9733-22ec71f57f3a, new version: 7
  A: 2856446a-34e6-435b-9fb5-f8f040bfa7ed, new version: 10
  A: 61ac3638-40a3-44b2-855a-985636ca985e, new version: 5
  A: ad84d445-b1ce-4377-82d9-7c633f28bf9a, new version: 5
  A: 56f2e9b5-4803-4e44-a0a4-a52dc79d57fe, new version: 3
  A: 622ecb68-fa81-4601-90b5-f8cd661e4520, new version: 6
  A: 11013227-0301-4a8c-b150-4db924484475, new version: 7
  A: 2772264c-6fb9-4d9d-9014-b416eed21254, new version: 5
  A: c8b150f0-0164-475b-a75e-74b47800a9ff, new version: 5
  A: afcce5ad-65de-4ed2-8516-5e093d3ac99a, new version: 11
  A: 54902e45-3467-49a4-8abc-529f2c8cfb80, new version: 6
  A: 61d29caf-6c15-4d1e-9ccb-7ad12ccc0bc7, new version: 1
  A: 14ed1aa9-ebfd-4cf9-a463-0ac59ec55204, new version: 3
  A: 5cd8e1f7-0050-4afc-b2df-904e40b2f5ae, new version: 4
  A: 16fac1a1-21ee-4ca6-b720-458e3855d046, new version: 3
  A: b9554892-5e0e-424b-83a0-5aef95aa43bf, new version: 3
  A: 15a8ba77-1c13-4274-88fe-6bd14133861e, new version: 3
  A: 58c6d58b-a0d3-412d-b3b8-0981a9400607, new version: 3
run `build-release --update-version-lock` to update version.lock.json and deprecated_rules.json
Package saved to: /detection-rules-fork/releases/8.2
- sha256: e8b3dc341ceca90781e42fc1e8a8d39661ff15c6946b773489da2fbe71348773
- 670 rules included
rm -rf dist
mkdir dist
cp -r releases/*/*.zip dist/
```

</details>

### CODE REVIEW

File: rules/windows/defense_evasion_posh_process_injection.toml [hunk 1]
```diff
@@ -2,6 +2,8 @@
 creation_date = "2021/10/14"
 maturity = "production"
 updated_date = "2022/03/15"
+min_stack_comments = "meh"
```
Comment (brokensound77) [inline rules/windows/defense_evasion_posh_process_injection.toml:5]: TODO: remove this before merging - for testing only

**Comment by brokensound77** [comment]
Opened as draft - still want to test a bit more to verify assumptions

File: rules/windows/defense_evasion_posh_process_injection.toml (hunk 1 above)
Comment (brokensound77) [reply rules/windows/defense_evasion_posh_process_injection.toml:5]:
```suggestion
```

File: rules/windows/defense_evasion_posh_process_injection.toml [hunk 2]
```diff
@@ -2,6 +2,8 @@
...
+min_stack_version = "7.15"
```
Comment (brokensound77) [inline rules/windows/defense_evasion_posh_process_injection.toml:6]:
```suggestion
```

**Review by Mikaayenson — APPROVED** [review]
WIP: LGTM

**Review by terrancedejesus — APPROVED** [review]
WIP: LGTM

**Pull request merged**